src/common/isolatable_hardwares.cpp

```diff
@@ -668,6 +668,65 @@ std::optional<std::vector<sdbusplus::message::object_path>>
     return listOfChildsInventoryPath;
 }
 
+std::optional<sdbusplus::message::object_path>
+    IsolatableHWs::getFRUInventoryPath(
+        const std::pair<LocationCode, InstanceId>& fruDetails,
+        const inv_path_lookup_func::LookupFuncForInvPath& fruInvPathLookupFunc)
+{
+    auto inventoryPathList = getInventoryPathsByLocCode(fruDetails.first);
+    if (!inventoryPathList.has_value())
+    {
+        return std::nullopt;
+    }
+
+    if (inventoryPathList->empty())
+    {
+        // The inventory object doesn't exist for the given location code.
```

Comment:
This seens to be real error case , logging error related to this?

Reply:
Planning to push separate PR to cover PEL use case for the guard operations.